docs/source/commands/contracts.mdx

```diff
@@ -0,0 +1,188 @@
+---
+title: Rover contract commands
+description: For use with Apollo Contract variants
+---
+
+A **contract** is a variant that is the result of filtering on a source variant:
+
+Rover commands that interact with contracts begin with `rover contract`.
+
+## Describing a contract variant
+
+These commands enable you to describe a contract in a federated graph.
+
+### `contract describe`
+
+TODO(caydie) is this correct? I'm assuming you always have to authn to rover with studio 
+> This command requires [authenticating Rover with Apollo Studio](../configuring/#authenticating-with-apollo-studio).
+
+You can use Rover to describe the contract variant of any source variant that Rover has access to.
+
+Run the `contract describe` command, like so:
+
+```bash
+rover contract describe my-graph@my-contract-variant
+```
+
+The argument `my-graph@my-contract-variant` in the example above is a [graph ref](../conventions/#graph-refs) that specifies the ID of the Studio graph you're fetching from, along with which [variant](/graphos/graphs/overview/#variants) you're fetching.
+
+The output will contain information of how the contract variant was configured such as the source variant, include and exclude tags.
+
+### `contract publish`
+
+> This command requires [authenticating Rover with Apollo Studio](../configuring/#authenticating-with-apollo-studio).
+
+You can use Rover to publish configuration changes to a contract.
+
+Use the `contract publish` command, like so:
+
+TODO(caydie) is it my graph or my supergraph and what would be a good example for the arguments?
+```bash
+rover contract publish my-graph@my-contract-variant \
+  --include [] \
+  --exclude [] \
+  --hide-unreachable-types
+```
+
+TODO(caydie): is it my-graph or is it my supergraph? i think it's my-graph
+The argument `my-graph@my-variant` in the example above is a [graph ref](../conventions/#graph-refs) that specifies the ID of the Studio graph you're publishing to, along with which [variant](/graphos/graphs/overview/#variants) you're publishing to.
+
+If the graph and source variant exists in the graph registry but the contract variant does _not_, a new contract variant is created on publish.
```

Comment:
maybe a silly question but how do you specify the source variant in this command? if i run the following:

```
rover contract publish my-graph@my-contract-variant \
  --include [] \
  --exclude [] \
  --hide-unreachable-types
```

it looks like i'm only specifying the contract variant - how do we know what the source variant should be in this scenario?

Reply:
you do indeed need to specify the `--source-variant` as an argument. we updated the doc to reflect that in the next commit!